src/server.mjs

```diff
@@ -265,6 +265,31 @@ async function getChunkedData(req, res, next) {
   console.log(`region = ${req.body.region}`);
   console.log(`tracks = ${JSON.stringify(req.body.tracks)}`);
 
+  // This will have a conitg, start, end, or a contig, start, distance
+  let parsedRegion;
+  try {
+    parsedRegion = parseRegion(req.body.region);
+  } catch (e) {
+    // Whatever went wrong in the parsing, it makes the request bad.
+    throw new BadRequestError("Wrong query: " + e.message + " See ? button above.");
+  }
+
+  // There's a chance this request was sent before the proper tracks were fetched
+  // This can happen when the bed file is a url and track names need to be downloaded
+  // TODO: There's also a chance the user decided to use different tracks than what was provided in the bedFile,
+  // maybe we shouldn't always replace tracks if found?
+  if (req.body.bedFile && isValidURL(req.body.bedFile)) {
```

Comment:
It looks like we're assuming that the client always will send the right tracks if their BED file is not a URL. I think we should run this check whether the BED is a URL or not.